Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +11,246 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var maxBalance = 0m;
+            var name = new StringBuilder();
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var balance = GetLargestBalance(peopleBalances);
+                GetNamesWithHighestBalance(ref name, ref maxBalance, peopleBalances, balance);
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            return $"{FixPlural(name.ToString())} had the most money ever. {maxBalanceNoComma}.";
+        }
+
+        private static decimal GetLargestBalance(string[] peopleBalances)
+        {
+            var maxBalance = 0m;
+
+            for (var i = 1; i < peopleBalances.Length; i++)
+            {
+                var isBalanceValid = decimal.TryParse(peopleBalances[i], out var balance);
+
+                if (isBalanceValid && balance > maxBalance)
+                {
+                    maxBalance = balance;
+                }
+            }
+
+            return maxBalance;
+        }
+
+        private static void GetNamesWithHighestBalance(ref StringBuilder name, ref decimal maxBalance, string[] peopleBalances, decimal balance)
```

Comment:
SetNames

Reply:
I renamed the method for GetNameWithHighest to SetNames, because I am internally setting the name in the method and not returning it.